kotlinx-coroutines-core/common/src/selects/Select.kt

```diff
@@ -844,7 +854,8 @@ internal open class SelectImplementation<R>(
 }
 
 private fun CancellableContinuation<Unit>.tryResume(onCancellation: ((cause: Throwable) -> Unit)?): Boolean {
```

Comment:
may be it makes sense to change `OnCancellationConstructor` to return lambda which is compatible with new `tryResume` to not allocate additional lambda here?